src/main/java/Duke.java

```diff
@@ -1,10 +1,352 @@
+        FileWrite.appendToFile(dataPath, DATA_SEPARATOR);
+        if (task.getStatus().equals("X")) {
+            FileWrite.appendToFile(dataPath, APPEND_DONE);
+        } else {
+            FileWrite.appendToFile(dataPath, APPEND_NOT_DONE);
+        }
+        FileWrite.appendToFile(dataPath, System.lineSeparator());
+    }
+
+
+    private static ArrayList<Task> readFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            int taskCount = 0;
+            File file = getFile();
+            Scanner s = new Scanner(file); // create a Scanner using the File as the source
+            while(s.hasNext()) {
+                String textLine = s.nextLine();
+                String[] inputs = textLine.split(DATA_SEPARATOR);
+                if(inputs[0].equals("T")) {
+                    taskCount = loadTodo(tasks, taskCount, inputs);
+                }
+                if(inputs[0].equals("D")) {
+                    taskCount = loadDeadline(tasks, taskCount, inputs);
+                }
+                if(inputs[0].equals("E")) {
+                    taskCount = loadEvent(tasks, taskCount, inputs);
+                }
+            }
+        } catch(FileNotFoundException fileNotFoundException) {
+            System.out.println("File not found!");
+        } catch(IOException ioexception) {
+            System.out.println("Theres a problem with making a new file!");
+        }
+        return tasks;
+    }
+
+    private static int loadEvent(ArrayList<Task> tasks, int taskCount, String[] inputs) {
+        tasks.add(new Event(inputs[1], inputs[2]));
+        taskCount++;
+        if(inputs[3].equals(APPEND_DONE)) {
+            tasks.get(taskCount - 1).markDone();
+        }
+        return taskCount;
+    }
+
+    private static int loadDeadline(ArrayList<Task> tasks, int taskCount, String[] inputs) {
+        tasks.add(new Deadline(inputs[1], inputs[2]));
+        taskCount++;
+        if(inputs[3].equals(APPEND_DONE)) {
+            tasks.get(taskCount - 1).markDone();
+        }
+        return taskCount;
+    }
+
+    private static int loadTodo(ArrayList<Task> tasks, int taskCount, String[] inputs) {
+        tasks.add(new Todo(inputs[1]));
+        taskCount++;
+        if(inputs[2].equals(APPEND_DONE)) {
+            tasks.get(taskCount - 1).markDone();
+        }
+        return taskCount;
+    }
+
+    private static File getFile() throws IOException {
+        File file = new File(FULL_RELATIVE_MEMORY_PATH);
+        if(!file.exists()) {
+            File makeNewDir = new File(DATA_DIRECTORY_PATH);
+            if(makeNewDir.mkdir()) {
+                System.out.println("Directory not found so I made a new directory!");
+            }
+            File NewFile = new File(FULL_RELATIVE_MEMORY_PATH);
+            if(NewFile.createNewFile()) {
+                System.out.println("File not found so I made a new file!");
+            }
+            return NewFile;
+        }
+        return file;
+    }
```

Comment:
Consider implementing a storage class to handle file creation and Parser class to handle inputs from file